snap/epoch.go

```diff
@@ -0,0 +1,288 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+)
+
+var (
+	ErrBadEpochExpression  = errors.New("invalid epoch expression")
+	ErrEpochOhSplat        = errors.New("0* is an invalid epoch")
+	ErrBadEpochNumber      = errors.New("epoch numbers must match [1-9][0-9]*")
```

Comment:
just realised this error message is wrong. Suggestions for how to fix it welcome :-)

Reply:
If nobody comes up with something better, I'll use a variation of what I wrote in Epoch's code comment (“Epoch numbers must be written as base 10 numbers, with no zero padding.”)

Reply:
Perhaps a minor tweak on that last suggestion:

_"epochs must be base 10 numbers with no leading zeros"_